### PR TITLE
Replace usage of foo.com with example.com in git_repository test data

### PR DESCRIPTION
**Pull request by edbaunton**
Fixes https://github.com/bazelbuild/bazel/issues/4409

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by iirina** [comment]
jenkins test this please

**Comment by ulfjack** [comment]
jenkins test this please

**Comment by ulfjack** [comment]
I get some errors in testing:
```
** test_git_repository *********************************************************
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
INFO: Starting clean.
Expected build/run to succeed
-- Test log: -----------------------------------------------------------
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
..........................
Loading: 
Loading: 0 packages loaded
Analyzing: target //planets:planet-info (2 packages loaded)
Analyzing: target //planets:planet-info (2 packages loaded)
Analyzing: target //planets:planet-info (5 packages loaded)
ERROR: /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/workspace/planets/BUILD:1:1: no such package '@pluto//': Invalid branch, tag, or commit: Ref 52f9a3f can not be resolved and referenced by '//planets:planet-info'
ERROR: Analysis of target '//planets:planet-info' failed; build aborted: no such package '@pluto//': Invalid branch, tag, or commit: Ref 52f9a3f can not be resolved
INFO: Elapsed time: 16.929s
FAILED: Build did NOT complete successfully (5 packages loaded)
ERROR: Build failed. Not running target
------------------------------------------------------------------------
test_git_repository FAILED: Expected regexp Pluto is a dwarf planet not found .
/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/bazel-out/darwin-fastbuild/bin/src/test/shell/bazel/git_repository_test.runfiles/io_bazel/src/test/shell/bazel/git_repository_test:95: in call to test_git_repository
INFO[git_repository_test 2018-02-14 16:45:17 (+0000)] Cleaning up workspace
FAILED: test_git_repository

** test_git_repository_and_nofetch *********************************************
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
INFO: Starting clean.
-- Test log: -----------------------------------------------------------
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
Loading: 
Loading: 0 packages loaded
ERROR: Skipping '@g//:g': no such package '@g//': Invalid branch, tag, or commit: Ref 22095302 can not be resolved
WARNING: Target pattern parsing failed.
ERROR: no such package '@g//': Invalid branch, tag, or commit: Ref 22095302 can not be resolved
INFO: Elapsed time: 0.408s
FAILED: Build did NOT complete successfully (0 packages loaded)
------------------------------------------------------------------------
test_git_repository_and_nofetch FAILED: Build failed .
/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/bazel-out/darwin-fastbuild/bin/src/test/shell/bazel/git_repository_test.runfiles/io_bazel/src/test/shell/bazel/git_repository_test:340: in call to test_git_repository_and_nofetch
INFO[git_repository_test 2018-02-14 16:45:26 (+0000)] Cleaning up workspace
FAILED: test_git_repository_and_nofetch

** test_git_repository_both_commit_tag_error ***********************************
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
INFO: Starting clean.
Expect run to fail.
INFO[git_repository_test 2018-02-14 16:45:31 (+0000)] Cleaning up workspace
PASSED: test_git_repository_both_commit_tag_error

** test_git_repository_no_commit_tag_error *************************************
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
INFO: Starting clean.
Expect run to fail.
INFO[git_repository_test 2018-02-14 16:45:36 (+0000)] Cleaning up workspace
PASSED: test_git_repository_no_commit_tag_error

** test_git_repository_not_refetched_on_server_restart *************************
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
INFO: Starting clean.
-- Test log: -----------------------------------------------------------
INFO: $TEST_TMPDIR defined: output root default is '/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa' and max_idle_secs default is '15'.
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Counting objects (0 / 0)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Counting objects (1 / 0)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (0 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (1 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (2 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (3 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (4 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (5 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (6 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (7 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (8 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (9 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (10 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Finding sources (10 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (0 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (1 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (2 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (3 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (4 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (5 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (6 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (7 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Getting sizes (7 / 7)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (0 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (70 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (140 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (208 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (241 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (304 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (367 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: remote: Compressing objects (367 / 367)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (0 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (1 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (2 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (3 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (4 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (5 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (6 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (7 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (8 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (9 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Receiving objects (10 / 10)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Resolving deltas (0 / 1)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Resolving deltas (1 / 1)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Updating references (0 / 1)
____Cloning /private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/_tmp/66cd6418c92c76af4424f5d2a65690aa/repos/refetch: Updating references (1 / 1)
ERROR: Skipping '@g//:g': no such package '@g//': Invalid branch, tag, or commit: Ref 22095302 can not be resolved.
WARNING: Target pattern parsing failed..
ERROR: no such package '@g//': Invalid branch, tag, or commit: Ref 22095302 can not be resolved.
____Elapsed time: 1.012s
------------------------------------------------------------------------
test_git_repository_not_refetched_on_server_restart FAILED: Build failed .
/private/var/netboot/tmp/tmpfs/tmp/bazel/execroot/io_bazel/bazel-out/darwin-fastbuild/bin/src/test/shell/bazel/git_repository_test.runfiles/io_bazel/src/test/shell/bazel/git_repository_test:276: in call to test_git_repository_not_refetched_on_server_restart
INFO[git_repository_test 2018-02-14 16:45:42 (+0000)] Cleaning up workspace
FAILED: test_git_repository_not_refetched_on_server_restart
```

**Comment by edbaunton** [comment]
That's odd because the tests definitely passed when I submitted the PR. Will investigate and fix.

**Comment by edbaunton** [comment]
Am I overlooking something obvious here?

```
$ bazel test //src/test/shell/bazel:git_repository_test
INFO: Analysed target //src/test/shell/bazel:git_repository_test (1 packages loaded).
INFO: Found 1 test target...
Target //src/test/shell/bazel:git_repository_test up-to-date:
  bazel-bin/src/test/shell/bazel/git_repository_test
INFO: Elapsed time: 72.356s, Critical Path: 71.80s
INFO: Build completed successfully, 5 total actions
//src/test/shell/bazel:git_repository_test                               PASSED in 71.4s

Executed 1 out of 1 test: 1 test passes.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
```

I don't see any files falling into my `.gitignore`. I am running on a macbook, so maybe a problem with a different platform?

**Comment by laszlocsomor** [comment]
@iirina is on vacation, i'll take this over.

This passes for me locally on Debian Linux with Bazel 0.11.0:
```
  $ bazel version
Build label: 0.11.0
Build target: bazel-out/k8-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Thu Dec 9 15:39:59 +50117 (1519402203599)
Build timestamp: 1519402203599
Build timestamp as int: 1519402203599


  $ git fetch origin-gh refs/pull/4522/head && git checkout FETCH_HEAD 
From https://github.com/bazelbuild/bazel
 * branch                  refs/pull/4522/head -> FETCH_HEAD
HEAD is now at fc43836298 Replace usage of foo.com with example.com in git_repository test data


  $ bazel test //src/test/shell/bazel:git_repository_test //src/test/shell/bazel:skylark_git_repository_test --test_output=errors
INFO: Analysed 2 targets (0 packages loaded).
INFO: Found 2 test targets...
INFO: Elapsed time: 0.379s, Critical Path: 0.00s
INFO: Build completed successfully, 1 total action
//src/test/shell/bazel:git_repository_test                      (cached) PASSED in 49.1s
//src/test/shell/bazel:skylark_git_repository_test              (cached) PASSED in 69.0s

Executed 0 out of 2 tests: 2 tests pass.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
```

(Cached, because I re-ran all commands now just to be able to paste them nicely.)

**Comment by laszlocsomor** [comment]
I'll import this now and run internal presubmits. @buchgr tells me that will also trigger Buildkite.

**Comment by laszlocsomor** [comment]
@edbaunton : how did you build the .tar.gz files in this PR? i don't know how I'd verify them

**Comment by edbaunton** [comment]
We should document the preferred way in https://github.com/bazelbuild/bazel/blob/master/src/test/shell/bazel/testdata/README.md
but I'm relatively sure I just ran `tar -zcvf` (I don't have the logs any more).

**Comment by laszlocsomor** [comment]
> We should document the preferred way in https://github.com/bazelbuild/bazel/blob/master/src/test/shell/bazel/testdata/README.md
but I'm relatively sure I just ran tar -zcvf (I don't have the logs any more).

Thanks! Could you do that please? Better yet, could you write a genrule or a script that can regenerate them? (It's OK if this non-hermetic; we're checking the built tarballs into the repo anyway.)

Could you please also rebuild the tarballs just to make sure?

Also, could you please rebase this PR on top of @bazelbuild/bazel / master ? That way our new CI system (Buildkite) picks it up and I can trigger a test on it.

**Comment by lfpino** [comment]
Hi @edbaunton, Bazel sheriff here, what's the status of this pull request? It hasn't been updated for more than a week so it'd be good to understand if it's still ongoing or safe to close. Thanks.

**Comment by edbaunton** [comment]
Rebased. Updated the README and also verified the tars are built correctly.

File: src/test/shell/bazel/git_repository_test.sh [hunk 1]
```diff
@@ -63,8 +63,8 @@ function set_up() {
 # //planets has a dependency on a target in the pluto Git repository.
 function test_git_repository() {
   local pluto_repo_dir=$TEST_TMPDIR/repos/pluto
-  # Commit 85b8224 corresponds to tag 1-build. See testdata/pluto.git_log.
-  local commit_hash="b87de93"
+  # Commit corresponds to tag 1-build. See testdata/pluto.git_log.
+  local commit_hash="52f9a3f"
```
Comment (laszlocsomor) [inline src/test/shell/bazel/git_repository_test.sh]: please use full commit hashes to prevent collisions (here and below)

**Comment by laszlocsomor** [comment]
Thanks! Presubmit tests are running. Once green, I'll import this change.

**Pull request closed without merging**